### ISSUE TITLE
Quota Exceeded Error on Android 4 and 5

### ISSUE THREAD

**Issue by laurenkyla** [pouchdb/pouchdb#5462]
### Environment (Node.js/browser/hybrid app/etc.)

Ionic (angular) 
### Browser/platform (Chrome/FF/Safari/Edge/iOS/Android/etc.)

Android 4 & 5
### Adapter (IndexedDB/WebSQL/LevelDB/etc.)

IndexedDB
### Server (CouchDB/Cloudant/Couchbase/PouchDB Server/etc.)

Cloudant
### Please describe your issue. Test cases (e.g. JSBin) are very helpful!

My app stopps working on older Android phones 4 & 5. Works fine on IOS, via the browser or newer Androids. 

App hangs on the inital login after it successfully syncs the database to it's own copy:

change{"direction":"pull","change":{"ok":true,"start_time":"2016-07-10T21:31:31.293Z","docs_read":100,"docs_written":100,"doc_write_failures":0,"errors":[], (followed by a lot of data and revisions)

 it then calls localDB.find to check users credentials. Sometimes it manages to check the users credentials and login in which case I start getting quota exceeded errors otherwise hangs at this point and I get timeout errors. 

The database that it is syncing is only 1MB. I have tried various things like specifying a different adapter 'websql' turning on auto compaction and specifying a size. Nothing seems to work. 

Uncaught TransactionInactiveError: Failed to execute 'getKey' on 'IDBIndex': The transaction is not active. pouchdb.min.js:7
Database has a global failure DOMError {message: "Transaction timed out due to inactivity.", name: "TimeoutError"} pouchdb.min.js:7
Database has a global failure DOMError {message: "Transaction timed out due to inactivity.", name: "TimeoutError"} pouchdb.min.js:7
Database has a global failure DOMError {message: "Transaction timed out due to inactivity.", name: "TimeoutError"} pouchdb.min.js:7
login error{"status":500,"name":"indexed_db_went_bad","message":"unknown","reason":"TimeoutError"} 

Any help would be greatly appreciated as I've hit a wall on this one. 

Thanks

Lauren

**Comment by nolanlawson** [comment]
Sometimes when you're developing, it's possible to exceed the storage on a phone simply because it's in heavy use during development. If you're not seeing the error in the wild, but you're seeing it during development, then that could be why.

One easy fix is to just clear all browser data (in Chrome or whatever browser this is) and then try again. If you're still seeing Quota Exceeded errors, then it could be because you are simply using too much quota, and unfortunately [there is no solution for that](www.html5rocks.com/en/tutorials/offline/quota-research/). 

OTOH the error you logged doesn't look like a quota error; it looks like some other error I've never seen before ("timed out due to inactivity"). Don't suppose you can make a test to reproduce? E.g. an Android "hello world" app or a simple web page that reproduces this error?

**Comment by nolanlawson** [comment]
Sometimes when you're developing, it's possible to exceed the storage on a phone simply because it's in heavy use during development. If you're not seeing the error in the wild, but you're seeing it during development, then that could be why.

One easy fix is to just clear all browser data (in Chrome or whatever browser this is) and then try again. If you're still seeing Quota Exceeded errors, then it could be because you are simply using too much quota, and unfortunately [there is no solution for that](www.html5rocks.com/en/tutorials/offline/quota-research/). 

OTOH the error you logged doesn't look like a quota error; it looks like some other error I've never seen before ("timed out due to inactivity"). Don't suppose you can make a test to reproduce? E.g. an Android "hello world" app or a simple web page that reproduces this error?

**Comment by laurenkyla** [comment]
This is an app in the wild that has stopped working. 

I've cleared every cache I can think of multiple times to no avail.

I also tried as a test connecting to cloudant from the app to sidestep local storage as option but it returns an empty dataset with no errors. If I do the same query via cloudant itself I get results. I am up to date with both pouchdb and pouchdb-find. 

return remoteDB.createIndex({
     index: {fields: ['name','uniquecode','email']}
	}).then(function() {
    return remoteDB.find({
	  selector: {name: 'users', uniquecode : code, email : email },
	}).then(function (result) {	
	console.log('got login' + JSON.stringify(result));
	  return result.docs;
	}).catch(function (err) {
	  // an error
	  console.log('get login fail' + err + JSON.stringify(result));
	});

I'll see if I can reproduce the timeout error with a simple app.

**Comment by nolanlawson** [comment]
Hi @laurenkyla, thanks for the details. Can you please also answer:

- is this a Cordova/PhoneGap/ReactNative app?
- is it regular WebView or Crosswalk WebView?
- by any chance do you have the user agent string of the devices that are crashing? Android 4 and 5 actually have a wide gamut of Chromium versions, from v30 all the way through v50.
- can you make a small test case to reproduce? A "hello world" app or something?

In general it kinda sounds like you are syncing a huge amount of data, which would explain the timeouts and the quota exceeded errors. It's quite possible that the database you're trying to sync is just too big. Keep in mind that even if it's only 1MB in CouchDB, it may be a different size in PouchDB.

**Comment by laurenkyla** [comment]
It's an Ionic App. 

I have tried with both Crosswalk and without, it made no difference. 

I don't have the agent string the analytics I'm using doesn't record them unfortunately. 

I tested with an empty database and it worked fine. It seems to be an issue with indexedDB as I get errors in firefox and android but not in Chrome or IOS. 

I reduced the size of the database there are only 1500 rows and no attachments.  I am no longer getting quata errors but pouchdb-find queries seem to just stall and tried with only selector or changing the order of them to the one that return the least first.

I am seeing Timeout errors and in firefox: 'Database has a global failure' or no error but the application just stalling after calling localDB.find

I had this problem before so switched to cloudant thinking it was my database host but it's back. 

With the websql adapter on android data is 13.77mb. 1.1mb in cloudant.
